src/Articulate/Controllers/MardownEditorApiController.cs

```diff
@@ -59,193 +68,168 @@ public class ParseImageResponse
+                return BadRequest(ModelState);                
+            }
+
+            IPublishedContent published = _umbracoHelper.Content(content.Id);
+            return Ok(new { url = published.Url() });
         }
+        
 
-        //private static void CleanFiles(MultipartFileStreamProvider multiPartRequest)
-        //{
-        //    foreach (var f in multiPartRequest.FileData)
-        //    {
-        //        System.IO.File.Delete(f.LocalFileName);
-        //    }
-        //}
-
-        //private ParseImageResponse ParseImages(string body, MultipartFileStreamProvider multiPartRequest, bool extractFirstImageAsProperty)
-        //{
-        //    var firstImage = string.Empty;
-        //    var bodyText = Regex.Replace(body, @"\[i:(\d+)\:(.*?)]", m =>
-        //    {
-        //        var index = m.Groups[1].Value.TryConvertTo<int>();
-        //        if (index)
-        //        {
-        //            //get the file at this index
-        //            var file = multiPartRequest.FileData[index.Result];
-
-        //            var rndId = Guid.NewGuid().ToString("N");
-
-        //            using (var stream = System.IO.File.OpenRead(file.LocalFileName))
-        //            {
-        //                var fileUrl = "articulate/" + rndId + "/" + file.Headers.ContentDisposition.FileName.TrimStart("\"").TrimEnd("\"");
-
-        //                _mediaFileManager.FileSystem.AddFile(fileUrl, stream);
-                        
-        //                var result = string.Format("![{0}]({1})",
-        //                    fileUrl,
-        //                    fileUrl
-        //                );
-
-        //                if (extractFirstImageAsProperty && string.IsNullOrEmpty(firstImage))
-        //                {
-        //                    firstImage = fileUrl;
-        //                    //in this case, we've extracted the image, we don't want it to be displayed
-        //                    // in the content too so don't return it.
-        //                    return string.Empty;
-        //                }
-
-        //                return result;
-        //            }
-        //        }
-
-        //        return m.Value;
-        //    });
-
-        //    return new ParseImageResponse { BodyText = bodyText, FirstImage = firstImage };
-        //}
+        private ParseImageResponse ParseImages(string body, IFormFileCollection formFiles, bool extractFirstImageAsProperty)
+        {
+            var firstImage = string.Empty;
+            var bodyText = Regex.Replace(body, @"\[i:(\d+)\:(.*?)]", m =>
+            {
+                var index = m.Groups[1].Value.TryConvertTo<int>();
+                if (index)
+                {
+                    //get the file at this index
+                    var file = formFiles[index.Result];
+
+                    var rndId = Guid.NewGuid().ToString("N");
+
+                    var stream = new MemoryStream();
```

Comment:
MemoryStream is IDisposable so should prob be in a using statement

Reply:
Updated - quick question. Do you have something in Visual Studio that helps tell you in the IDE UI that something is IDisposable so that you should try to use a Usings?

Reply:
Sort of. IDisposable is a tricky one. You can run code analysis in VS and it might tell you about not disposing things. The issue is that it isn't required to use a 'using' clause and you can dispose things manually and in some cases that is required.